staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -285,6 +300,8 @@ func (s preparedGenericAPIServer) NonBlockingRun(stopCh <-chan struct{}) error {
 	// ensure cleanup.
 	go func() {
 		<-stopCh
+		glog.Infof("Shutting down nonblocking genericapiserver")
+		s.RunPreStopHooks()
```

Comment:
why this call **plus** the one in `Run`? I would only expect the later. This go func is only to synchronize the internalStopCh with the external one.

Reply:
I will remove